{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook identifies outliers in ambient and low pH conditions throughout the experiment\n",
    "\n",
    "The Fisher's Exact Test was used to identify outliers between pairs of buckets\n",
    "\n",
    "The CMH test was used to look for consistent allele frequency across biological replicates \n",
    "\n",
    "The sync file was generated from the  using Final_SyncFileGeneration_OpenAccess.ipynb\n",
    "\n",
    "I follow along with this tutorial: https://sourceforge.net/p/popoolation2/wiki/Tutorial/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#Calculate allele frequency changes using snp-freq-diff.pl\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/\n",
    "\n",
    "perl ./popoolation2_1201/snp-frequency-diff.pl --input /scratch/mbitter/ExomeCapture_Scratch/PoPoolation/Transcriptome/SyncFiles/Sync_1.sync --output-prefix Embr_D6Full_all --min-count 5 --min-coverage 50 --max-coverage 400\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "##Run Fisher's Exact Test to identify outliers using every possible pairwise comparisson (each day, treatment, etc.)\n",
    "##For the paper, the only lists extracted from this test were the outlier lists generated by comparing the day 0 larval populations to the larval populations in each treatment on days 6\n",
    "\n",
    "module load gcc/6.2.0\n",
    "module load perl/5.24.0\n",
    "\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/\n",
    "\n",
    "\n",
    "perl ./popoolation2_1201/fisher-test.pl --input ./PoPoolation/Transcriptome/SyncFiles/Sync_1.sync --output ./PoPoolation/Transcriptome/Outputs/all_2.fet --min-count 15 --min-coverage 50 --max-coverage 400 --suppress-noninformative"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading sync file and writing temporary R output file\n",
      "Calling R, to calculate the Cochran-Mantel-Haenszel test statistic\n",
      "Parsing R-output and writing output file\n",
      "Done\n",
      "Reading sync file and writing temporary R output file\n",
      "Calling R, to calculate the Cochran-Mantel-Haenszel test statistic\n",
      "Parsing R-output and writing output file\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "%%sh\n",
    "#CMH test to identify outliers between the fastest and slowest growers in ambient and low pH conditions\n",
    "module load gcc/6.2.0\n",
    "module load perl/5.24.0\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/\n",
    "\n",
    "#For Ambient pH replicates\n",
    "\n",
    "perl ./popoolation2_1201/cmh-test.pl --input ./PoPoolation/Transcriptome/SyncFiles/Sync_1.sync --output Embr_D6SizeSep_HighpH_3.cmh --min-count 5 --min-coverage 50 --max-coverage 400 --population 8-12,10-14\n",
    "\n",
    "\n",
    "#For Low pH replicates\n",
    "perl ./popoolation2_1201/cmh-test.pl --input ./PoPoolation/Transcriptome/SyncFiles/Sync_1.sync --output Embr_D6SizeSep_LowpH_3.cmh --min-count 5 --min-coverage 50 --max-coverage 400 --population 9-13,11-15\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading sync file and writing temporary R output file\n",
      "Calling R, to calculate the Cochran-Mantel-Haenszel test statistic\n",
      "Parsing R-output and writing output file\n",
      "Done\n",
      "Reading sync file and writing temporary R output file\n",
      "Calling R, to calculate the Cochran-Mantel-Haenszel test statistic\n",
      "Parsing R-output and writing output file\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "%%sh\n",
    "\n",
    "##Now comparing the Day 0 larval population (embryos) to the Day 25 larval populations in each treatment using the CMH test\n",
    "\n",
    "module load gcc/6.2.0\n",
    "module load perl/5.24.0\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/\n",
    "\n",
    "#For High pH replicates\n",
    "\n",
    "perl ./popoolation2_1201/cmh-test.pl --input ./PoPoolation/Transcriptome/SyncFiles/Sync_1.sync --output Embr_D25_HighpH_3.cmh --min-count 5 --min-coverage 50 --max-coverage 400 --population 1-3,1-5,1-6\n",
    "\n",
    "\n",
    "#For Low pH replicates\n",
    "perl ./popoolation2_1201/cmh-test.pl --input ./PoPoolation/Transcriptome/SyncFiles/Sync_1.sync --output Embr_D25_LowpH_3.cmh --min-count 5 --min-coverage 50 --max-coverage 400 --population 1-2,1-4,1-7\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading sync file and writing temporary R output file\n",
      "Calling R, to calculate the Cochran-Mantel-Haenszel test statistic\n",
      "Parsing R-output and writing output file\n",
      "Done\n",
      "Reading sync file and writing temporary R output file\n",
      "Calling R, to calculate the Cochran-Mantel-Haenszel test statistic\n",
      "Parsing R-output and writing output file\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "%%sh\n",
    "\n",
    "#Finally, comparing the Day 0 larval population to the settled larvae in each treatment using the CMH test\n",
    "\n",
    "module load gcc/6.2.0\n",
    "module load perl/5.24.0\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/\n",
    "\n",
    "#For High pH replicates\n",
    "\n",
    "perl ./popoolation2_1201/cmh-test.pl --input ./PoPoolation/Transcriptome/SyncFiles/Sync_1.sync --output emb_settled_HighpH_2.cmh --min-count 5 --min-coverage 50 --max-coverage 400 --population 1-19,1-21\n",
    "\n",
    "\n",
    "#For Low pH replicates\n",
    "perl ./popoolation2_1201/cmh-test.pl --input ./PoPoolation/Transcriptome/SyncFiles/Sync_1.sync --output emb_settled_LowpH_2.cmh --min-count 5 --min-coverage 50 --max-coverage 400 --population 1-18,1-20,1-22\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
