{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This script conducts raw read QC/trimming and variant calling on the pools (larvae) from PE_3 using only the original transcriptome used to create the exome capture array. This is to reduce computational time and develop a picture of genetic variation change throughout the experiment. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "FastQC on RawReads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#load modules\n",
    "module load java-jdk/1.8.0_92\n",
    "module load fastqc/0.11.5\n",
    "\n",
    "\n",
    "fastqc -o /scratch/mbitter/ExomeCapture_Scratch/Pools_FastQC_RawReads --extract /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/*.fastq.gz  -t 8\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "View files using MultiQC (actually viewed by downloading html output to local drive and using browser)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "#Load modules\n",
    "module load gcc/6.2.0\n",
    "module load python/2.7.13\n",
    "\n",
    "multiqc -n raw_Pools_PE3_multiqc /scratch/mbitter/ExomeCapture_Scratch/Pools_FastQC_RawReads/ -o /scratch/mbitter/ExomeCapture_Scratch/MultiQC_RawReads/\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This looks pretty good. Areas of consideration are sequence duplication levels, and potentially high levels of adapter sequences (which will be addressed in the next step).\n",
    "\n",
    "\n",
    "Now read trimming using trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#load modules\n",
    "module load java-jdk/1.8.0_92\n",
    "module load trimmomatic/0.36\n",
    "\n",
    "#Trim adapters (this is the standard Illumina TrueSeq3 adapters), and low quality reads\n",
    "\n",
    "\n",
    "#Run on all files\n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-31_S1_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-31_S1_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-31_S1_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-31_S1_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-31_S1_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-31_S1_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-38_S2_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-38_S2_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-38_S2_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-38_S2_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-38_S2_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-38_S2_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-39_S3_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-39_S3_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-39_S3_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-39_S3_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-39_S3_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-39_S3_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-40_S4_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-40_S4_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-40_S4_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-40_S4_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-40_S4_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-40_S4_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-41_S5_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-41_S5_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-41_S5_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-41_S5_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-41_S5_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-41_S5_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-42_S6_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-42_S6_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-42_S6_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-42_S6_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-42_S6_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-42_S6_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-43_S7_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-43_S7_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-43_S7_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-43_S7_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-43_S7_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-43_S7_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-44_S8_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-44_S8_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-44_S8_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-44_S8_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-44_S8_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-44_S8_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-45_S9_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-45_S9_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-45_S9_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-45_S9_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-45_S9_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-45_S9_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-47_S10_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-47_S10_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-47_S10_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-47_S10_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-47_S10_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-47_S10_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-49_S11_L007_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl2-MB-49_S11_L007_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-49_S11_L007_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-49_S11_L007_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-49_S11_L007_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-49_S11_L007_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-32_S12_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-32_S12_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-32_S12_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-32_S12_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-32_S12_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-32_S12_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-33_S13_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-33_S13_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-33_S13_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-33_S13_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-33_S13_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-33_S13_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-34_S14_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-34_S14_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-34_S14_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-34_S14_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-34_S14_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-34_S14_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-35_S15_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-35_S15_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-35_S15_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-35_S15_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-35_S15_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-35_S15_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-36_S16_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-36_S16_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-36_S16_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-36_S16_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-36_S16_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-36_S16_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-37_S17_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-37_S17_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-37_S17_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-37_S17_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-37_S17_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-37_S17_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-51_S18_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-51_S18_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-51_S18_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-51_S18_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-51_S18_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-51_S18_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-52_S19_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-52_S19_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-52_S19_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-52_S19_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-52_S19_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-52_S19_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-53_S20_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-53_S20_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-53_S20_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-53_S20_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-53_S20_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-53_S20_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-55_S21_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-55_S21_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-55_S21_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-55_S21_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-55_S21_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-55_S21_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "java -jar $TRIMMOMATIC PE -threads 8 -phred33 /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-56_S22_L008_R1_001.fastq.gz /group/pfister-lab/Bitter/ExomeCapture/Pools_RawSequences201806/CP-MB-pl3-MB-56_S22_L008_R2_001.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-56_S22_L008_R1_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-56_S22_L008_R1_UP_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-56_S22_L008_R2_trimmomatic.fastq.gz /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-56_S22_L008_R2_UP_trimmomatic.fastq.gz ILLUMINACLIP:/scratch/mbitter/ExomeCapture_Scratch/TruSeqAdapters.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                       \n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Re-run fastqc and multiqc on trimmed reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#Load modules\n",
    "\n",
    "module purge\n",
    "module load java-jdk/1.10.0_92\n",
    "module load fastqc/0.11.7\n",
    "\n",
    "fastqc -o /scratch/mbitter/ExomeCapture_Scratch/Pools_FastQC_TrimmedReads/ --extract /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/*.fastq.gz -t 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "#Load modules\n",
    "module load gcc/6.2.0\n",
    "module load python/2.7.13\n",
    "\n",
    "multiqc -n trimmomatic_Pools_PE3_multiqc /scratch/mbitter/ExomeCapture_Scratch/Pools_FastQC_TrimmedReads/ -o /scratch/mbitter/ExomeCapture_Scratch/MultiQC_Trimmomatic/\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Adapter content issue is resolved, sequence duplication levels still an issue for some, but will be taken care of during de-duplication."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "#Load modules\n",
    "module purge\n",
    "module load gcc/6.2.0 bowtie2\n",
    "which bowtie2\n",
    "\n",
    "#Align all pool read files to the transcriptome, and same index that was generated for the adult alignment\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-31_S1_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-31_S1_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-31_S1_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-31_S1_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-31_S1_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-31_S1_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-38_S2_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-38_S2_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-38_S2_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-38_S2_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-38_S2_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-38_S2_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-39_S3_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-39_S3_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-39_S3_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-39_S3_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-39_S3_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-39_S3_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_AlignmentMgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-40_S4_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-40_S4_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-40_S4_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-40_S4_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-40_S4_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-40_S4_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-41_S5_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-41_S5_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-41_S5_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-41_S5_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-41_S5_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-41_S5_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-42_S6_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-42_S6_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-42_S6_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-42_S6_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-42_S6_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-42_S6_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-43_S7_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-43_S7_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-43_S7_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-43_S7_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-43_S7_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-43_S7_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-44_S8_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-44_S8_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-44_S8_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-44_S8_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-44_S8_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-44_S8_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-45_S9_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-45_S9_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-45_S9_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-45_S9_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-45_S9_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-45_S9_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-47_S10_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-47_S10_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-47_S10_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-47_S10_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-47_S10_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-47_S10_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-49_S11_L007_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-49_S11_L007_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-49_S11_L007_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl2-MB-49_S11_L007_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-49_S11_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl2-MB-49_S11_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_AlignmentMgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-32_S12_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-32_S12_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-32_S12_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-32_S12_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-32_S12_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-32_S12_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-33_S13_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-33_S13_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-33_S13_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-33_S13_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-33_S13_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-33_S13_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-34_S14_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-34_S14_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-34_S14_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-34_S14_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-34_S14_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-34_S14_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-35_S15_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-35_S15_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-35_S15_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-35_S15_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-35_S15_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-35_S15_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-36_S16_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-36_S16_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-36_S16_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-36_S16_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-36_S16_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-36_S16_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-37_S17_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-37_S17_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-37_S17_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-37_S17_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-37_S17_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-37_S17_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-51_S18_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-51_S18_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-51_S18_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-51_S18_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-51_S18_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-51_S18_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-52_S19_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-52_S19_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-52_S19_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-52_S19_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-52_S19_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReadsCP-MB-pl3-MB-52_S19_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-53_S20_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-53_S20_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-53_S20_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-53_S20_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-53_S20_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-53_S20_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-55_S21_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-55_S21_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-55_S21_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-55_S21_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-55_S21_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-55_S21_transcriptome_bt2.stout\n",
    "bowtie2 --local -x /gpfs/data/pfister-lab/Bitter/ExomeCapture/Adult_Transcriptome_Alignment/MgalloTranscriptomeIndex -1 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-56_S22_L008_R1_trimmomatic.fastq.gz  -2 /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-56_S22_L008_R2_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-56_S22_L008_R1_UP_trimmomatic.fastq.gz -U /scratch/mbitter/ExomeCapture_Scratch/Pools_Trimmomatic_Reads/CP-MB-pl3-MB-56_S22_L008_R2_UP_trimmomatic.fastq.gz -S /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-56_S22_transcriptome_bt2.sam -p 14 2>/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/CP-MB-pl3-MB-56_S22_transcriptome_bt2.stout\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/\n",
    "\n",
    "module load gcc intel\n",
    "module load samtools\n",
    "\n",
    "samtools view -S -b CP-MB-pl2-MB-40_S4_transcriptome_bt2.sam > CP-MB-pl2-MB-40_S4_transcriptome_bt2.bam\n",
    "samtools view -S -b CP-MB-pl3-MB-32_S13_transcriptome_bt2.sam > CP-MB-pl3-MB-32_S13_transcriptome_bt2.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generate the .bam files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "module purge\n",
    "module load gcc intel\n",
    "module load samtools\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads\n",
    "\n",
    "for i in /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/*.sam; do\n",
    "    samtools view -S -b $i > ${i/.sam/.bam}\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sort the .bam files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "module load gcc intel\n",
    "module load samtools\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads\n",
    "\n",
    "for i in /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/*.bam; do\n",
    "    samtools sort $i -o ${i/.bam/.sorted.bam}\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All .sorted.bam files were de-duped using the PoolsTranscriptome_MarkDups.sh in the /FinalScripts directory"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generate depth table for each .sorted.deduped.bam file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "module load gcc intel\n",
    "module load samtools\n",
    "\n",
    "/scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/\n",
    "\n",
    "for i in /scratch/mbitter/ExomeCapture_Scratch/PoolsTranscriptomeAlignment/MappedReads/*.sorted.deduped.bam;do\n",
    "    samtools depth $i > ${i/.sorted.deduped.bam/DepthTable.txt}\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generate a new file with average coverage in each sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_Transcriptome_DepthSummStats\n",
    "\n",
    "touch PoolsDepthSummary.txt\n",
    "for file in *.txt;do \n",
    "    avg_depth=$(awk '{sum += $3; n ++} END {print sum/n}' ${file})\n",
    "    echo ${file} ${avg_depth} >> PoolsDepthSummary.txt\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I then added reading groups to all of the sorted, deduped .bam files from the command line. \n",
    "The scripts for doing this are located in /FinalScripts/AddReadGroupScripts/Pools_AddReadGroups"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Index bam files for GATK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_Transcriptome_BamFiles/\n",
    "\n",
    "for i in /scratch/mbitter/ExomeCapture_Scratch/Pools_Transcriptome_BamFiles/*.RG.bam;do\n",
    "    samtools index $i \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reordering the .bam files according to reference fasta to ensure that they are in the same order as the adult vcf file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_BamsForHapCalling\n",
    "\n",
    "module load java-jdk/1.8.0_92\n",
    "module load picard/2.8.1\n",
    "\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-31_S1_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-31_S1_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-38_S2_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-38_S2_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-39_S3_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-39_S3_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-40_S4_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-40_S4_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-41_S5_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-41_S5_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-42_S6_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-42_S6_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-43_S7_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-43_S7_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-44_S8_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-44_S8_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-45_S9_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-45_S9_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-47_S10_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-47_S10_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl2-MB-49_S11_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl2-MB-49_S11_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-32_S12_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-32_S12_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-33_S13_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-33_S13_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-34_S14_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-34_S14_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-35_S15_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-35_S15_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-36_S16_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-36_S16_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-37_S17_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-37_S17_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-51_S18_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-51_S18_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-52_S19_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-52_S19_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-53_S20_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-53_S20_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-55_S21_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-55_S21_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "java -Xmx6G -XX:ParallelGCThreads=4 -jar ${PICARD} ReorderSam I=CP-MB-pl3-MB-56_S22_transcriptome_bt2.sorted.deduped.RG.bam O=CP-MB-pl3-MB-56_S22_transcriptome_bt2.sorted.deduped.RG.reordered.bam R=/gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta CREATE_INDEX=TRUE\n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As I ran into trouble concatenating all separate scaffold Adult Variant .vcf files into one and running GATK, I chose to run estimate the frequency of each variant in each pooled .bam file using each separate .vcf file and will concatenate all the resulting .vcf's for each pooled sample after"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample31\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample31_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample31_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample31_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample31_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-31_S1_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o Sample31_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample32\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample32_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample32_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample32_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample32_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-32_S12_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample32/Sample32_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample33\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample33_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample33_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample33_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample33_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-33_S13_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample33/Sample33_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample34\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample34_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample34_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample34_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample34_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-34_S14_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample34/Sample34_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample35\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample35_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample35_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample35_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample35_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-35_S15_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample35/Sample35_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample36\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample36_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample36_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample36_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample36_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-36_S16_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample36/Sample36_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample37\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample37_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample37_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample37_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample37_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-37_S17_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample37/Sample37_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample38\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample38_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample38_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample38_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample38_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-38_S2_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample38/Sample38_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample39\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample39_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample39_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample39_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample39_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-39_S3_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample39/Sample39_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample40\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample40_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample40_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample40_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample40_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-40_S4_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample40/Sample40_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample41\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample41_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample41_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample41_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample41_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-41_S5_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample41/Sample41_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample42\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample42_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample42_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample42_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample42_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-42_S6_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample42/Sample42_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample43\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample43_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample43_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample43_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample43_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-43_S7_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample43/Sample43_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample44\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample44_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample44_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample44_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample44_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-44_S8_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample44/Sample44_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample45\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample45_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample45_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample45_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample45_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-45_S9_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample45/Sample45_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample47\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample47_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample47_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample47_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample47_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-47_S10_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample47/Sample47_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample49\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample49_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample49_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample49_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample49_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl2-MB-49_S11_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample49/Sample49_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample51\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample51_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample51_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample51_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample51_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-51_S18_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample51/Sample51_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample52\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample52_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample52_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample52_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample52_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-52_S19_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample52/Sample52_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample53\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample53_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample53_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample53_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample53_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-53_S20_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample53/Sample53_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample55\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample55_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample55_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample55_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample55_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-55_S21_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample55/Sample55_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "#Sample56\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Adult_Transcriptome_FilteredVCFs\")\n",
    "\n",
    "nFiles = 122\n",
    "\n",
    "for d in range(1,nFiles+1):\n",
    "    sh = open(\"Sample56_HapCaller_TranscriptomeScaffold\"+str(d)+\".sh\",\"w\")\n",
    "    sh.write(\"#PBS -N Sample56_HapCaller_TranscriptomeScaffold\"+str(d)+\\\n",
    "            \"\\n#PBS -S /bin/bash\"+\\\n",
    "            \"\\n#PBS -l nodes=1:ppn=1\")\n",
    "    sh.write(\"\\n#PBS -l walltime=100:00:00\\n\")\n",
    "    sh.write(\"#PBS -l mem=6gb\")\n",
    "    sh.write(\"\\n#PBS -o Sample56_HappCaller_TranscriptomeScaffold\"+str(d)+\".out\"+\\\n",
    "            \"\\n#PBS -e Sample56_HapCaller_TranscriptomeScaffold\"+str(d)+\".err\"+\\\n",
    "            \"\\n\\nmodule load java-jdk/1.8.0_92\"+\\\n",
    "            \"\\nmodule load gatk/3.8\"+\\\n",
    "            \"\\n\\ncd /scratch/mbitter/ExomeCapture_Scratch/\"+\\\n",
    "            \"\\n\\njava -Xmx6G -jar ${GATK} -T HaplotypeCaller -R /gpfs/data/pfister-lab/Bitter/PublishedSequences/Mgallo_TranscriptomeForVariantCalling/Mgalloprovincialis_transcripts_final.fasta -I ./Pools_Transcriptome_BamFiles/CP-MB-pl3-MB-56_S22_transcriptome_bt2.sorted.deduped.RG.reordered.bam --genotyping_mode GENOTYPE_GIVEN_ALLELES -alleles ./Adult_Transcriptome_FilteredVCFs/UnifiedGenotyper_TranscriptomeScaffolds\"+str(d)+\".recode.vcf -o ./Pools_GATK/Sample56/Sample56_TranscriptomeScaffold\"+str(d)+\".vcf\")\n",
    "    sh.close"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merge all VCF's from each sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample31/VCFs\n",
    "bcftools concat *.vcf -o Sample31_TranscriptomeVariants.vcf\n",
    "scp Sample31_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample32/VCFs\n",
    "bcftools concat *.vcf -o Sample32_TranscriptomeVariants.vcf\n",
    "scp Sample32_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample33/VCFs\n",
    "bcftools concat *.vcf -o Sample33_TranscriptomeVariants.vcf\n",
    "scp Sample33_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample34/VCFs\n",
    "bcftools concat *.vcf -o Sample34_TranscriptomeVariants.vcf\n",
    "scp Sample34_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample35/VCFs\n",
    "bcftools concat *.vcf -o Sample35_TranscriptomeVariants.vcf\n",
    "scp Sample35_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample36/VCFs\n",
    "bcftools concat *.vcf -o Sample36_TranscriptomeVariants.vcf\n",
    "scp Sample36_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample37/VCFs\n",
    "bcftools concat *.vcf -o Sample37_TranscriptomeVariants.vcf\n",
    "scp Sample37_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample38/VCFs\n",
    "bcftools concat *.vcf -o Sample38_TranscriptomeVariants.vcf\n",
    "scp Sample38_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample39/VCFs\n",
    "bcftools concat *.vcf -o Sample39_TranscriptomeVariants.vcf\n",
    "scp Sample39_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample40/VCFs\n",
    "bcftools concat *.vcf -o Sample40_TranscriptomeVariants.vcf\n",
    "scp Sample40_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample41/VCFs\n",
    "bcftools concat *.vcf -o Sample41_TranscriptomeVariants.vcf\n",
    "scp Sample41_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample42/VCFs\n",
    "bcftools concat *.vcf -o Sample42_TranscriptomeVariants.vcf\n",
    "scp Sample42_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample43/VCFs\n",
    "bcftools concat *.vcf -o Sample43_TranscriptomeVariants.vcf\n",
    "scp Sample43_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample44/VCFs\n",
    "bcftools concat *.vcf -o Sample44_TranscriptomeVariants.vcf\n",
    "scp Sample44_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample45/VCFs\n",
    "bcftools concat *.vcf -o Sample45_TranscriptomeVariants.vcf\n",
    "scp Sample45_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample47/VCFs\n",
    "bcftools concat *.vcf -o Sample47_TranscriptomeVariants.vcf\n",
    "scp Sample47_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample49/VCFs\n",
    "bcftools concat *.vcf -o Sample49_TranscriptomeVariants.vcf\n",
    "scp Sample49_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample51/VCFs\n",
    "bcftools concat *.vcf -o Sample51_TranscriptomeVariants.vcf\n",
    "scp Sample51_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample52/VCFs\n",
    "bcftools concat *.vcf -o Sample52_TranscriptomeVariants.vcf\n",
    "scp Sample52_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample53/VCFs\n",
    "bcftools concat *.vcf -o Sample53_TranscriptomeVariants.vcf\n",
    "scp Sample53_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample55/VCFs\n",
    "bcftools concat *.vcf -o Sample55_TranscriptomeVariants.vcf\n",
    "scp Sample55_TranscriptomeVariants.vcf ../../MergedVCFs\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample56/VCFs\n",
    "bcftools concat *.vcf -o Sample56_TranscriptomeVariants.vcf\n",
    "scp Sample56_TranscriptomeVariants.vcf ../../MergedVCFs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/Sample49/VCFs\n",
    "bcftools concat *.vcf -o Sample49_TranscriptomeVariants.vcf\n",
    "scp Sample49_TranscriptomeVariants.vcf ../../MergedVCFs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Filter all concatenated VCF files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/MergedVCFs\n",
    "\n",
    "for i in ./*.vcf;do\n",
    "    vcftools --vcf $i --maf 0.05 --max-maf 0.95 --minQ 30 --minDP 30 --maxDP 350 --recode --recode-INFO-all --out ${i/.vcf}\n",
    "done\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generate matrices for each sample identifying allelic depth at each variable site. This is a proxy for the alternate and reference allele frequency "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs\n",
    "for i in *.recode.vcf.gz;do\n",
    "    bcftools query -f '%CHROM\\t%POS\\t[ %AD{0}]\\t[ %AD{1}]\\n' $i > ${i/.recode.vcf.gz/.csv}\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create the final data matrix. This will consist of rows of variants, and columns of samples. Values in each sell correspond to the allele frequency at the ith variant in the jth sample. Variant sites in which a sample(s) does not have data are still included as NA's. These will be removed during downstream analyses."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample43_TranscriptomeVariants.csv\n",
      "Sample31_TranscriptomeVariants.csv\n",
      "Sample53_TranscriptomeVariants.csv\n",
      "Sample51_TranscriptomeVariants.csv\n",
      "Sample33_TranscriptomeVariants.csv\n",
      "Sample44_TranscriptomeVariants.csv\n",
      "Sample56_TranscriptomeVariants.csv\n",
      "Sample38_TranscriptomeVariants.csv\n",
      "Sample42_TranscriptomeVariants.csv\n",
      "Sample40_TranscriptomeVariants.csv\n",
      "Sample35_TranscriptomeVariants.csv\n",
      "Sample52_TranscriptomeVariants.csv\n",
      "Sample45_TranscriptomeVariants.csv\n",
      "Sample34_TranscriptomeVariants.csv\n",
      "Sample36_TranscriptomeVariants.csv\n",
      "Sample49_TranscriptomeVariants.csv\n",
      "Sample32_TranscriptomeVariants.csv\n",
      "Sample39_TranscriptomeVariants.csv\n",
      "Sample37_TranscriptomeVariants.csv\n",
      "Sample41_TranscriptomeVariants.csv\n",
      "Sample47_TranscriptomeVariants.csv\n",
      "Sample55_TranscriptomeVariants.csv\n"
     ]
    }
   ],
   "source": [
    "import pandas\n",
    "import glob, os\n",
    "import functools\n",
    "\n",
    "\n",
    "\n",
    "#Function to generate a two column data frame for each sample: \n",
    "#one column listing the variants, the other the frequency of each variant within the sample\n",
    "def get_af_df(f):\n",
    "    \"\"\"\n",
    "    takes in a file f (generated above)\n",
    "    calculates allele frequencies, depth, and generates variant column. \n",
    "    It returns a two column data frame with a column of variants\n",
    "    and a column of the alternate allele frequency of each variant within the sample.\n",
    "    \"\"\"\n",
    "    data = pandas.read_csv(f, sep = '\\t', header=None, names = [\"Gene\", \"Position\", \"Ref\", \"Alt\"])\n",
    "    data['Depth'] = data['Ref']+data['Alt'] \n",
    "    data['RefFreq'] = data['Ref']/data['Depth']\n",
    "    data['AltFreq'] = data['Alt']/data['Depth']\n",
    "    data['Variant'] = data['Gene'].astype(str) + '_' + data['Position'].astype(str)\n",
    "    data = data[['Gene', 'Position', 'Variant', 'Ref', 'Alt', 'Depth', 'RefFreq', 'AltFreq']]\n",
    "\n",
    "    sample = f.split(\"_\")[0] \n",
    "    data[sample] = data[\"AltFreq\"] #This is keeping the column of AF's associated with the file it came from\n",
    "    af_df = data[['Variant', sample]].copy() #The af_df has a column of variant names and alternate allele frequencies (titled by sample name)\n",
    "    \n",
    "    return(af_df)  \n",
    "\n",
    "\n",
    "\n",
    "#### Run the Function on all samples and merge the resulting datasets to obtain the final allele freq matrix ####\n",
    "\n",
    "os.chdir(\"/scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/AFTables/\")\n",
    "\n",
    "files = glob.glob('*.csv') #a list of the sample file names\n",
    "\n",
    "# Run the get_af_df function on the first file of the list\n",
    "af_df = get_af_df(files[0])\n",
    "Tables = [af_df] #Add the output to Tables (a list of dataframes) to store each individual output\n",
    "\n",
    "print(files[0])\n",
    "\n",
    "# Read in the 2nd file and onwwards and add each new file to Tables\n",
    "for f in files[1:]:\n",
    "\n",
    "    print(f) #Printing out the file name to ensure that it is running correctly\n",
    "    \n",
    "    #the next allele frequency dataframe\n",
    "    next_af_df = get_af_df(f)\n",
    "    Tables.append(next_af_df)\n",
    "    \n",
    "    # intersect the DataFrames to only keep variants that are shared among all samples\n",
    "    af_df = af_df.merge(next_af_df, how=\"outer\", on=\"Variant\")   \n",
    "    \n",
    "\n",
    "    \n",
    "#Save final matrix as .csv  \n",
    "af_df.to_csv(\"/scratch/mbitter/ExomeCapture_Scratch/Pools_GATK/AFTables/Pools_Transcriptome_AFMatrix.csv\", index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant</th>\n",
       "      <th>Sample43</th>\n",
       "      <th>Sample31</th>\n",
       "      <th>Sample53</th>\n",
       "      <th>Sample51</th>\n",
       "      <th>Sample33</th>\n",
       "      <th>Sample44</th>\n",
       "      <th>Sample56</th>\n",
       "      <th>Sample38</th>\n",
       "      <th>Sample42</th>\n",
       "      <th>...</th>\n",
       "      <th>Sample45</th>\n",
       "      <th>Sample34</th>\n",
       "      <th>Sample36</th>\n",
       "      <th>Sample49</th>\n",
       "      <th>Sample32</th>\n",
       "      <th>Sample39</th>\n",
       "      <th>Sample37</th>\n",
       "      <th>Sample41</th>\n",
       "      <th>Sample47</th>\n",
       "      <th>Sample55</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CL2Contig5_All_453</td>\n",
       "      <td>0.906977</td>\n",
       "      <td>0.895425</td>\n",
       "      <td>0.896552</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.840580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.880597</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.870968</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CL6Contig7_All_1473</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.673469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.652174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.593750</td>\n",
       "      <td>0.700000</td>\n",
       "      <td>0.716418</td>\n",
       "      <td>0.740741</td>\n",
       "      <td>...</td>\n",
       "      <td>0.676923</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.728395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.707692</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.652174</td>\n",
       "      <td>0.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CL6Contig7_All_1486</td>\n",
       "      <td>0.232558</td>\n",
       "      <td>0.322034</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.416667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.255814</td>\n",
       "      <td>0.283019</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.295775</td>\n",
       "      <td>...</td>\n",
       "      <td>0.385542</td>\n",
       "      <td>0.351351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.259615</td>\n",
       "      <td>0.361111</td>\n",
       "      <td>0.293333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.163934</td>\n",
       "      <td>0.189655</td>\n",
       "      <td>0.384615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CL6Contig7_All_1620</td>\n",
       "      <td>0.156682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.185185</td>\n",
       "      <td>0.144860</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.201794</td>\n",
       "      <td>0.175824</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.185366</td>\n",
       "      <td>...</td>\n",
       "      <td>0.190244</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.180791</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.185567</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.149573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CL6Contig7_All_1629</td>\n",
       "      <td>0.610879</td>\n",
       "      <td>0.512690</td>\n",
       "      <td>0.678363</td>\n",
       "      <td>0.677130</td>\n",
       "      <td>0.691943</td>\n",
       "      <td>0.722689</td>\n",
       "      <td>0.665025</td>\n",
       "      <td>0.640244</td>\n",
       "      <td>0.685106</td>\n",
       "      <td>...</td>\n",
       "      <td>0.689189</td>\n",
       "      <td>0.695312</td>\n",
       "      <td>0.51145</td>\n",
       "      <td>0.634703</td>\n",
       "      <td>0.784615</td>\n",
       "      <td>0.631944</td>\n",
       "      <td>0.708134</td>\n",
       "      <td>0.528169</td>\n",
       "      <td>0.654822</td>\n",
       "      <td>0.710843</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Variant  Sample43  Sample31  Sample53  Sample51  Sample33  \\\n",
       "0   CL2Contig5_All_453  0.906977  0.895425  0.896552       NaN       NaN   \n",
       "1  CL6Contig7_All_1473  0.600000  0.673469       NaN  0.652174       NaN   \n",
       "2  CL6Contig7_All_1486  0.232558  0.322034       NaN  0.416667       NaN   \n",
       "3  CL6Contig7_All_1620  0.156682       NaN  0.185185  0.144860       NaN   \n",
       "4  CL6Contig7_All_1629  0.610879  0.512690  0.678363  0.677130  0.691943   \n",
       "\n",
       "   Sample44  Sample56  Sample38  Sample42    ...     Sample45  Sample34  \\\n",
       "0       NaN  0.840580       NaN       NaN    ...          NaN  0.880597   \n",
       "1  0.593750  0.700000  0.716418  0.740741    ...     0.676923       NaN   \n",
       "2  0.255814  0.283019  0.200000  0.295775    ...     0.385542  0.351351   \n",
       "3  0.201794  0.175824       NaN  0.185366    ...     0.190244  0.142857   \n",
       "4  0.722689  0.665025  0.640244  0.685106    ...     0.689189  0.695312   \n",
       "\n",
       "   Sample36  Sample49  Sample32  Sample39  Sample37  Sample41  Sample47  \\\n",
       "0       NaN       NaN       NaN       NaN  0.870968       NaN       NaN   \n",
       "1       NaN  0.728395       NaN  0.707692       NaN  0.714286  0.652174   \n",
       "2       NaN  0.259615  0.361111  0.293333       NaN  0.163934  0.189655   \n",
       "3       NaN       NaN  0.180791       NaN  0.185567       NaN       NaN   \n",
       "4   0.51145  0.634703  0.784615  0.631944  0.708134  0.528169  0.654822   \n",
       "\n",
       "   Sample55  \n",
       "0       NaN  \n",
       "1  0.750000  \n",
       "2  0.384615  \n",
       "3  0.149573  \n",
       "4  0.710843  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "af_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generating .sync file from pool vcfs to use poolSeq and PoPoolation outlier identification software\n",
    "\n",
    "First, merge all pooled samples' vcfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#First bgzip all files\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs/\n",
    "\n",
    "for file in ./*.recode.vcf;do\n",
    "    bgzip $file\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#Create index files for all files using samtools tabix\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs/\n",
    "\n",
    "for file in ./*.gz;do\n",
    "    tabix -p vcf $file\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs/\n",
    "\n",
    "bcftools merge -m id Sample31_TranscriptomeVariants.recode.vcf.gz Sample32_TranscriptomeVariants.recode.vcf.gz Sample33_TranscriptomeVariants.recode.vcf.gz Sample34_TranscriptomeVariants.recode.vcf.gz Sample35_TranscriptomeVariants.recode.vcf.gz Sample36_TranscriptomeVariants.recode.vcf.gz Sample37_TranscriptomeVariants.recode.vcf.gz Sample38_TranscriptomeVariants.recode.vcf.gz Sample39_TranscriptomeVariants.recode.vcf.gz Sample40_TranscriptomeVariants.recode.vcf.gz Sample41_TranscriptomeVariants.recode.vcf.gz Sample42_TranscriptomeVariants.recode.vcf.gz Sample43_TranscriptomeVariants.recode.vcf.gz Sample44_TranscriptomeVariants.recode.vcf.gz Sample45_TranscriptomeVariants.recode.vcf.gz Sample47_TranscriptomeVariants.recode.vcf.gz Sample49_TranscriptomeVariants.recode.vcf.gz Sample51_TranscriptomeVariants.recode.vcf.gz Sample52_TranscriptomeVariants.recode.vcf.gz Sample53_TranscriptomeVariants.recode.vcf.gz Sample55_TranscriptomeVariants.recode.vcf.gz Sample56_TranscriptomeVariants.recode.vcf.gz -o Pools_TranscriptomeVariants.vcf\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use this to generate the appropriate bcftools output file to ultimately generate the .sync file necessary for poolSeq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/poolSeq_Files/\n",
    "\n",
    "bcftools query -f '%CHROM\\t%POS\\t%REF\\t%ALT[\\t%AD{0}:%AD{1}]\\n' Pools_TranscriptomeVariants.vcf.gz -o Pools_ADTableMerged.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Pools_ADTableMerged.txt file was then transferred to my local drive and used in  AlleleFrequencyAnalysis_poolSeq.Rmd "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Calculating basic summary statistics on pooled samples\n",
    "\n",
    "\n",
    "First, I will calculate nucleotide diversity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#Using 1000 bp sliding window\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs\n",
    "\n",
    "for i in ./*.gz;do\n",
    "    vcftools --gzvcf $i --window-pi 1000 --out ${i/.recode.vcf.gz}\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob, os\n",
    "import functools\n",
    "\n",
    "\n",
    "os.chdir(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_PopGenStats/Pi\")\n",
    "\n",
    "files = glob.glob(*.pi)\n",
    "\n",
    "for f in files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Sample        pi\n",
      "0   Sample34  0.003277\n",
      "1   Sample40  0.003217\n",
      "2   Sample37  0.003347\n",
      "3   Sample35  0.003397\n",
      "4   Sample52  0.003396\n",
      "5   Sample56  0.003305\n",
      "6   Sample49  0.003569\n",
      "7   Sample44  0.003409\n",
      "8   Sample55  0.003222\n",
      "9   Sample47  0.003303\n",
      "10  Sample41  0.003351\n",
      "11  Sample43  0.003323\n",
      "12  Sample38  0.003381\n",
      "13  Sample42  0.003349\n",
      "14  Sample36  0.003340\n",
      "15  Sample53  0.003309\n",
      "16  Sample45  0.003352\n",
      "17  Sample51  0.003414\n",
      "18  Sample32  0.003178\n",
      "19  Sample33  0.003424\n",
      "20  Sample39  0.003334\n",
      "21  Sample31  0.003462\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import glob, os\n",
    "import functools\n",
    "\n",
    "\n",
    "def get_pi(f):\n",
    "    \n",
    "    \"\"\"\n",
    "    Takes in file (generated above)\n",
    "    Calculates exome-wide pi\n",
    "    Returns sample name and pi value\n",
    "    \"\"\"\n",
    "    \n",
    "    data = pd.read_csv(f, sep = '\\t')\n",
    "    pi = data[\"PI\"].mean()\n",
    "    sample = f.split(\"_\")[0]\n",
    "    pi_df = (sample, pi)\n",
    "\n",
    "    return(pi_df)\n",
    "\n",
    "\n",
    "##Run function on all samples to obtain output file of samples and corresponding nucleotide diversity\n",
    "\n",
    "os.chdir(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_PopGenStats/Pi\")\n",
    "\n",
    "Table= []\n",
    "#generate list of file names\n",
    "files = glob.glob(\"*.pi\")\n",
    "\n",
    "\n",
    "for f in files:\n",
    "    next_pi = get_pi(f)\n",
    "    Table.append(next_pi)\n",
    "    \n",
    "Table = pd.DataFrame(Table, columns = ['Sample', 'pi'])\n",
    "\n",
    "#Table.pivot(index = 0, columns = 1, values = 2)\n",
    "print(Table)\n",
    "\n",
    "Table.to_csv(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_PopGenStats/Pi/Pools_Transcriptome_Pi.csv\", index = False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating new pool VCF files with less stringent quality filtering, but more stringent minimum depth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#Note: I created a new directory /MergedVCFs_2 for the newly\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs_2/\n",
    "\n",
    "for i in ./*.vcf;do \n",
    "    vcftools --vcf $i --maf 0.01 --max-maf 0.99 --minQ 30 --minDP 50 --maxDP 400 --recode --recode-INFO-all --out ${i/.vcf/_2}\n",
    "done\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generate allelic depth matrices for each sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs_2/\n",
    "\n",
    "for i in ./*.recode.vcf;do\n",
    "    bcftools query -f '%CHROM\\t%POS\\t[ %AD{0}]\\t[ %AD{1}]\\n' $i > ${i/.recode.vcf/.csv}\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample34_TranscriptomeVariants_2.csv\n",
      "Sample43_TranscriptomeVariants_2.csv\n",
      "Sample53_TranscriptomeVariants_2.csv\n",
      "Sample52_TranscriptomeVariants_2.csv\n",
      "Sample39_TranscriptomeVariants_2.csv\n",
      "Sample56_TranscriptomeVariants_2.csv\n",
      "Sample55_TranscriptomeVariants_2.csv\n",
      "Sample38_TranscriptomeVariants_2.csv\n",
      "Sample36_TranscriptomeVariants_2.csv\n",
      "Sample47_TranscriptomeVariants_2.csv\n",
      "Sample51_TranscriptomeVariants_2.csv\n",
      "Sample33_TranscriptomeVariants_2.csv\n",
      "Sample42_TranscriptomeVariants_2.csv\n",
      "Sample35_TranscriptomeVariants_2.csv\n",
      "Sample32_TranscriptomeVariants_2.csv\n",
      "Sample31_TranscriptomeVariants_2.csv\n",
      "Sample49_TranscriptomeVariants_2.csv\n",
      "Sample45_TranscriptomeVariants_2.csv\n",
      "Sample44_TranscriptomeVariants_2.csv\n",
      "Sample37_TranscriptomeVariants_2.csv\n",
      "Sample41_TranscriptomeVariants_2.csv\n",
      "Sample40_TranscriptomeVariants_2.csv\n"
     ]
    }
   ],
   "source": [
    "import pandas\n",
    "import glob, os\n",
    "import functools\n",
    "\n",
    "\n",
    "\n",
    "#Function to generate a two column data frame for each sample: \n",
    "#one column listing the variants, the other the frequency of each variant within the sample\n",
    "def get_af_df(f):\n",
    "    \"\"\"\n",
    "    takes in a file f (generated above)\n",
    "    calculates allele frequencies, depth, and generates variant column. \n",
    "    It returns a two column data frame with a column of variants\n",
    "    and a column of the alternate allele frequency of each variant within the sample.\n",
    "    \"\"\"\n",
    "    data = pandas.read_csv(f, sep = '\\t', header=None, names = [\"Gene\", \"Position\", \"Ref\", \"Alt\"])\n",
    "    data['Depth'] = data['Ref']+data['Alt'] \n",
    "    data['RefFreq'] = data['Ref']/data['Depth']\n",
    "    data['AltFreq'] = data['Alt']/data['Depth']\n",
    "    data['Variant'] = data['Gene'].astype(str) + '_' + data['Position'].astype(str)\n",
    "    data = data[['Gene', 'Position', 'Variant', 'Ref', 'Alt', 'Depth', 'RefFreq', 'AltFreq']]\n",
    "\n",
    "    sample = f.split(\"_\")[0] \n",
    "    data[sample] = data[\"AltFreq\"] #This is keeping the column of AF's associated with the file it came from\n",
    "    af_df = data[['Variant', sample]].copy() #The af_df has a column of variant names and alternate allele frequencies (titled by sample name)\n",
    "    \n",
    "    return(af_df)  \n",
    "\n",
    "\n",
    "\n",
    "#### Run the Function on all samples and merge the resulting datasets to obtain the final allele freq matrix ####\n",
    "\n",
    "os.chdir(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs_2/\")\n",
    "\n",
    "files = glob.glob('*.csv') #a list of the sample file names\n",
    "\n",
    "# Run the get_af_df function on the first file of the list\n",
    "af_df = get_af_df(files[0])\n",
    "Tables = [af_df] #Add the output to Tables (a list of dataframes) to store each individual output\n",
    "\n",
    "print(files[0])\n",
    "\n",
    "# Read in the 2nd file and onwwards and add each new file to Tables\n",
    "for f in files[1:]:\n",
    "\n",
    "    print(f) #Printing out the file name to ensure that it is running correctly\n",
    "    \n",
    "    #the next allele frequency dataframe\n",
    "    next_af_df = get_af_df(f)\n",
    "    Tables.append(next_af_df)\n",
    "    \n",
    "    # intersect the DataFrames to only keep variants that are shared among all samples\n",
    "    af_df = af_df.merge(next_af_df, how=\"outer\", on=\"Variant\")   \n",
    "    \n",
    "\n",
    "    \n",
    "#Save final matrix as .csv  \n",
    "af_df.to_csv(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs_2/Pools_Transcriptome_AFMatrix_2.csv\", index = False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, begin steps toward generating new sync files for downstream analysis with PoPoolation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#First bgzip all files\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs_2/\n",
    "\n",
    "for file in ./*.recode.vcf;do\n",
    "    bgzip $file\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#Create index files for all files using samtools tabix\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs_2/\n",
    "\n",
    "for file in ./*.gz;do\n",
    "    tabix -p vcf $file\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "#Merge all vcf's\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs_2/\n",
    "\n",
    "bcftools merge -m id Sample31_TranscriptomeVariants_2.recode.vcf.gz Sample32_TranscriptomeVariants_2.recode.vcf.gz Sample33_TranscriptomeVariants_2.recode.vcf.gz Sample34_TranscriptomeVariants_2.recode.vcf.gz Sample35_TranscriptomeVariants_2.recode.vcf.gz Sample36_TranscriptomeVariants_2.recode.vcf.gz Sample37_TranscriptomeVariants_2.recode.vcf.gz Sample38_TranscriptomeVariants_2.recode.vcf.gz Sample39_TranscriptomeVariants_2.recode.vcf.gz Sample40_TranscriptomeVariants_2.recode.vcf.gz Sample41_TranscriptomeVariants_2.recode.vcf.gz Sample42_TranscriptomeVariants_2.recode.vcf.gz Sample43_TranscriptomeVariants_2.recode.vcf.gz Sample44_TranscriptomeVariants_2.recode.vcf.gz Sample45_TranscriptomeVariants_2.recode.vcf.gz Sample47_TranscriptomeVariants_2.recode.vcf.gz Sample49_TranscriptomeVariants_2.recode.vcf.gz Sample51_TranscriptomeVariants_2.recode.vcf.gz Sample52_TranscriptomeVariants_2.recode.vcf.gz Sample53_TranscriptomeVariants_2.recode.vcf.gz Sample55_TranscriptomeVariants_2.recode.vcf.gz Sample56_TranscriptomeVariants_2.recode.vcf.gz -o Pools_TranscriptomeVariants_2.vcf\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "\n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/MergedVCFs_2/\n",
    "\n",
    "bcftools query -f '%CHROM\\t%POS\\t%REF\\t%ALT[\\t%AD{0}:%AD{1}]\\n' Pools_TranscriptomeVariants_2.vcf -o Pools_ADTableMerged_2.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "Generating two new data frames from _2 vcf files to use in Arjun's geodist visualization software\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "#Using #_2.vcf files (same file used for PoPoolation analyses) to generate alternate allele count dataframe (ACMatrix) \n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/VCFs_2/\n",
    "\n",
    "for i in ./*.recode.vcf.gz;do\n",
    "    bcftools query -f '%CHROM\\t%POS[\\t%AD{1}]\\n' $i > ${i/.recode.vcf.gz/_AC.txt}\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample33_TranscriptomeVariants_2_AC.txt\n",
      "Sample42_TranscriptomeVariants_2_AC.txt\n",
      "Sample51_TranscriptomeVariants_2_AC.txt\n"
     ]
    }
   ],
   "source": [
    "import pandas\n",
    "import glob, os\n",
    "import functools\n",
    "\n",
    "\n",
    "\n",
    "#Function to generate a three column data frame for each sample: \n",
    "#one column listing the contig, one the position, and the third the frequency of each variant within the sample\n",
    "def get_ac_df(f):\n",
    "    \"\"\"\n",
    "    takes in a file f (generated above)\n",
    "    It returns a three column data frame with a column of variants\n",
    "    and a column of the alternate allele count of each variant within the sample.\n",
    "    \"\"\"\n",
    "    data = pandas.read_table(f, sep = '\\t', header=None, names = [\"Gene\", \"Position\", \"AC\"])\n",
    "    #data = data[['Gene', 'Position', 'AC']]\n",
    "\n",
    "    sample = f.split(\"_\")[0] \n",
    "    data[sample] = data['AC'] #This is keeping the column of AF's associated with the file it came from\n",
    "    ac_df = data[['Gene', 'Position', 'AC', sample]].copy() #The af_df has a column of variant names and alternate allele frequencies (titled by sample name)\n",
    "    \n",
    "    return(ac_df)  \n",
    "\n",
    "\n",
    "\n",
    "#### Run the Function on all samples and merge the resulting datasets to obtain the final allele freq matrix ####\n",
    "\n",
    "os.chdir(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/ACTables_2/\")\n",
    "\n",
    "files = glob.glob('*.txt') #a list of the sample file names\n",
    "\n",
    "# Run the get_af_df function on the first file of the list\n",
    "ac_df = get_ac_df(files[0])\n",
    "Tables = [ac_df] #Add the output to Tables (a list of dataframes) to store each individual output\n",
    "\n",
    "print(files[0])\n",
    "\n",
    "# Read in the 2nd file and onwwards and add each new file to Tables\n",
    "for f in files[1:]:\n",
    "\n",
    "    print(f) #Printing out the file name to ensure that it is running correctly\n",
    "    \n",
    "    #the next allele frequency dataframe\n",
    "    next_ac_df = get_ac_df(f)\n",
    "    Tables.append(next_ac_df)\n",
    "    \n",
    "    # intersect the DataFrames to only keep variants that are shared among all samples\n",
    "    ac_df = ac_df.merge(next_ac_df, how=\"outer\", on=\"Position\")   \n",
    "    \n",
    "\n",
    "    \n",
    "#Save final matrix as .csv  \n",
    "ac_df.to_csv(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/VCFs_2/Pools_Transcriptome_ACMatrix_2.csv\", index = False)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%sh\n",
    "#Using #_2.vcf files (same file used for PoPoolation analyses) to generate depth dataframe (Depth_Matrix) \n",
    "cd /gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/VCFs_2/\n",
    "\n",
    "for i in ./*.recode.vcf.gz;do\n",
    "    bcftools query -f '%CHROM\\t%POS\\t%REF\\t%ALT\\t%DP\\n' $i > ${i/.recode.vcf.gz/_DP.txt}\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas\n",
    "import glob, os\n",
    "import functools\n",
    "\n",
    "\n",
    "\n",
    "#Function to generate a three column data frame for each sample: \n",
    "#one column listing the contig, one the position, and the third the frequency of each variant within the sample\n",
    "def get_dp_df(f):\n",
    "    \"\"\"\n",
    "    takes in a file f (generated above)\n",
    "    calculates allele frequencies, depth, and generates variant column. \n",
    "    It returns a two column data frame with a column of variants\n",
    "    and a column of the alternate allele frequency of each variant within the sample.\n",
    "    \"\"\"\n",
    "    data = pandas.read_table(f, sep = '\\t', header=None, names = [\"Gene\", \"Position\", \"Ref\", \"Alt\", \"DP\"])\n",
    "    #data = data[['Gene', 'Position', 'Variant', 'Ref', 'Alt', 'Depth', 'RefFreq', 'AltFreq']]\n",
    "\n",
    "    sample = f.split(\"_\")[0] \n",
    "    data[sample] = data[\"DP\"] #This is keeping the column of AF's associated with the file it came from\n",
    "    dp_df = data[['Gene','Position', sample]].copy() #The dp_df has a columns of genes, position, and depth (titled by sample name)\n",
    "    \n",
    "    return(dp_df)  \n",
    "\n",
    "\n",
    "\n",
    "#### Run the Function on all samples and merge the resulting datasets to obtain the final allele freq matrix ####\n",
    "\n",
    "os.chdir(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/DPTables_2/\")\n",
    "\n",
    "files = glob.glob('*.txt') #a list of the sample file names\n",
    "\n",
    "# Run the get_af_df function on the first file of the list\n",
    "dp_df = get_dp_df(files[0])\n",
    "Tables = [dp_df] #Add the output to Tables (a list of dataframes) to store each individual output\n",
    "\n",
    "print(files[0])\n",
    "\n",
    "# Read in the 2nd file and onwwards and add each new file to Tables\n",
    "for f in files[1:]:\n",
    "\n",
    "    print(f) #Printing out the file name to ensure that it is running correctly\n",
    "    \n",
    "    #the next allele frequency dataframe\n",
    "    next_dp_df = get_dp_df(f)\n",
    "    Tables.append(next_dp_df)\n",
    "    \n",
    "    # intersect the DataFrames to only keep variants that are shared among all samples\n",
    "    dp_df = dp_df.merge(next_dp_df, how=\"outer\", on=\"DP\")   \n",
    "    \n",
    "\n",
    "    \n",
    "#Save final matrix as .csv  \n",
    "dp_df.to_csv(\"/gpfs/data/pfister-lab/Bitter/ExomeCapture/Pools_GATK/Pools_Transcriptome_DPMatrix_2.csv\", index = False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
